floris/floris_model.py

```diff
@@ -1577,7 +1577,7 @@ def set_operation_model(self, operation_model: str | List[str]):
 
     def copy(self):
         """Create an independent copy of the current FlorisModel object"""
-        return FlorisModel(self.core.as_dict())
+        return self.__class__(self.core.as_dict(), **self.secondary_init_kwargs)
```

Comment:
I would note here in the docstring that `self.__class__` is used so that subclasses will return themselves. It's clear what you did when I come to this from the pull request, but I think it would be odd if I came to this without the added context. I love the idea of this fwiw.